### ISSUE TITLE
feat: Read Multiple Files DuckDB

### ISSUE THREAD

**Issue by jayceslesar** [ibis-project/ibis#4906]
### Is your feature request related to a problem?

I can take a look at this but seems like the _generate_view_code Regex Dispatcher might need to be robustified to be able to take a list of regexes, assert that they are all the same and provide the same functionality. Alternatively we could assume that the elements in the list all follow the same general pattern and will point to the same place but I don't really like that.

This is definitely a nice to have as I wouldn't think there to be a speed difference here? But would be really interesting if there was speed difference (multiple connections vs one vs maybe even threading the connection to each file downstream)

### Describe the solution you'd like

We can see [here](https://duckdb.org/docs/data/parquet) that DuckDB can take in a list of exact paths, which seems much cleaner than 

```py
import ibis

con = ibis.duckdb.connect()

# partitions is a list of my files
t = con.register(partitions[0])
for path in partitions[1:]:
    t = t.union(con.register(path))
```

Would be nice to just:

```py
import ibis

con = ibis.duckdb.connect()

# partitions is a list of my files
t = con.register(partitions)
```

### What version of ibis are you running?

Latest Master

### What backend(s) are you using, if any?

DuckDB

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Thanks @jayceslesar, yeah this would be really nice to have.

**Comment by cpcloud** [comment]
This was addressed in 4.1:

```
…/ibis on  deferred-docs is 📦 v4.1.0 via 🐍 v3.10.9 via ❄️  impure (ibis-3.10.9-env)
❯ ipython

In [1]: import ibis

In [2]: t = ibis.read_parquet('ci/ibis-testing-data/parquet/functional_alltypes/*.parq')

In [3]: t.head().execute()
Out[3]:
     id  bool_col  tinyint_col  smallint_col  int_col  bigint_col  float_col  double_col date_string_col string_col           timestamp_col  year  month
0  5770      True            0             0        0           0        0.0         0.0     b'08/01/10'       b'0' 2010-08-01 00:00:00.000  2010      8
1  5771     False            1             1        1          10        1.1        10.1     b'08/01/10'       b'1' 2010-08-01 00:01:00.000  2010      8
2  5772      True            2             2        2          20        2.2        20.2     b'08/01/10'       b'2' 2010-08-01 00:02:00.100  2010      8
3  5773     False            3             3        3          30        3.3        30.3     b'08/01/10'       b'3' 2010-08-01 00:03:00.300  2010      8
4  5774      True            4             4        4          40        4.4        40.4     b'08/01/10'       b'4' 2010-08-01 00:04:00.600  2010      8

In [4]: from pathlib import Path

In [5]: t = ibis.read_parquet([f for f in Path('ci/ibis-testing-data/parquet/functional_alltypes').glob('*.parq')])

In [6]: t.head().execute()
Out[6]:
     id  bool_col  tinyint_col  smallint_col  int_col  bigint_col  float_col  double_col date_string_col string_col           timestamp_col  year  month
0  5770      True            0             0        0           0        0.0         0.0     b'08/01/10'       b'0' 2010-08-01 00:00:00.000  2010      8
1  5771     False            1             1        1          10        1.1        10.1     b'08/01/10'       b'1' 2010-08-01 00:01:00.000  2010      8
2  5772      True            2             2        2          20        2.2        20.2     b'08/01/10'       b'2' 2010-08-01 00:02:00.100  2010      8
3  5773     False            3             3        3          30        3.3        30.3     b'08/01/10'       b'3' 2010-08-01 00:03:00.300  2010      8
4  5774      True            4             4        4          40        4.4        40.4     b'08/01/10'       b'4' 2010-08-01 00:04:00.600  2010      8
```
